core/src/main/java/com/graphhopper/routing/util/parsers/TagParser.java

```diff
@@ -33,4 +34,10 @@
     void createEncodedValues(EncodedValueLookup lookup, List<EncodedValue> registerNewEncodedValue);
 
     IntsRef handleWayTags(IntsRef edgeFlags, ReaderWay way, boolean ferry, IntsRef relationFlags);
+
+    // todo: this way subclasses have to overwrite both methods in case they need to make use of custom areas, but on the
+    // otherhand existing subclasses can remain unchanged...
+    default IntsRef handleWayTags(IntsRef edgeFlags, ReaderWay way, boolean ferry, IntsRef relationFlags, List<CustomArea> customAreas) {
+        return handleWayTags(edgeFlags, way, ferry, relationFlags);
+    }
 }
```

Comment:
We should rather remove the old method instead of adding another one, and actually no signature change is necessary if we put the custom areas into an artificial way tag (see above).